DSCResources/cProjectDeploy/cProjectDeploy.psm1

```diff
@@ -0,0 +1,102 @@
+function Invoke-InitialDeploy{
+    param (
+        [Parameter(Mandatory=$True)]
+        [string]$name,
+        [Parameter(Mandatory=$True)]
+        [string]$apiKey,
+        [Parameter(Mandatory=$True)]
+        [string]$octopusServerUrl,
+        [Parameter(Mandatory=$True)]
+        [string]$Project,
+        [Parameter(Mandatory=$True)]
+        [string[]]$Environments,
+        [string]$Version = "latest",
+        [Switch]$Wait = $true
+    )
+
+    $octoDL = "http://download.octopusdeploy.com/octopus-tools/2.5.10.39/OctopusTools.2.5.10.39.zip"
+    <#if (Test-Path "$($env:SystemDrive)\Octopus\OctopusTools\$($Project)_initial.txt"){
+        Write-Verbose "Initial Deployment for $Project already done, nothing to do"
+        return
+    }#>
+
+    if ( -not (Test-Path "$($env:SystemDrive)\Octopus\OctopusTools\Octo.exe")){
+        if ( -not (Test-Path "$($env:SystemDrive)\Octopus\OctopusTools.zip")){
+            Write-Verbose "Downloading OctopusTools from $octoDL"
+            Request-File -url $octoDL -saveAs "$($env:SystemDrive)\Octopus\OctopusTools.zip"
+        }
+        Write-Verbose "Unpacking OctopusTools to $($env:SystemDrive)\Octopus\OctopusTools\"
+        Add-Type -assemblyname System.IO.Compression.FileSystem
+        [System.IO.Compression.ZipFile]::ExtractToDirectory("$($env:SystemDrive)\Octopus\OctopusTools.zip","$($env:SystemDrive)\Octopus\OctopusTools\")
+
+    }
+    pushd "$($env:SystemDrive)\Octopus\OctopusTools\"
+    foreach ($environment in $environments){
+        Write-Verbose "Deploying Project $Project to environment $environment"
+        $deployArguments = @("deploy-release", "--project", $Project, "--deployto", $environment, "--releaseNumber", $Version, "--specificmachines", $env:COMPUTERNAME, "--server", $octopusServerUrl, "--apiKey", $apiKey)
+        if ($Wait){
+            $deployArguments += "--waitfordeployment"
+        }
+        Invoke-AndAssert { & .\octo.exe $deployArguments}
```

Comment:
Invoke-AndAssert is a function in the cTentacleAgent resource-module so will not exist in the execution context of this module hence will break the Set-TargetResource at this point. Probably easiest to just copy that function over to here.